webproject/src/main/java/com/google/sps/dataManagers/UserRoomManager.java

```diff
@@ -0,0 +1,61 @@
+package com.google.sps.dataManagers;
+
+import java.util.Optional;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+import com.google.sps.data.UserRoom;
+import com.google.sps.util.FirebaseUtil;
+
+@Singleton
+public class UserRoomManager {
+    private FirebaseUtil firebaseUtil;
+
+    @Inject
+    public UserRoomManager(FirebaseUtil firebaseUtil) {
+        this.firebaseUtil = firebaseUtil;
+    }
+
+    public void addUserRoom(String userEmail, String roomId) {
+        UserRoom userRoom = UserRoom.newBuilder().setUserEmail(userEmail).setRoomId(roomId).setUserEmailRoom().build();
+
+        firebaseUtil.getUserRoomReference().push()
+                .setValue(userRoom, (databaseError, databaseReference) -> {
+                    if (databaseError != null) {
+                        System.out.println("Data could not be saved " + databaseError.getMessage());
+                    } else {
+                        System.out.println("Data saved successfully.");
+                    }
+                });
+    }
+
+    public Optional getUserRoom(String userEmail, String roomId) throws InterruptedException {
```

Comment:
wrap the exception thrown to become ServletException with some message explaining the errors

Reply:
https://www.journaldev.com/1973/servlet-exception-and-error-handling-example-tutorial#:~:text=to%20the%20user.-,Servlet%20Exception%20and%20Error%20Handling,useful%20HTML%20response%20to%20user.

Reply:
okay!